src/credentials.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  CredentialDefinition,
+  CredentialOffer,
+  CredentialManifestDisplayMapping,
+  ClaimEntry,
+} from "@jolocom/protocol-ts"
+import { jsonpath } from "./util"
+
+export interface DisplayVal {
+  label?: string
+  key?: string
+  value?: string
+}
+
+export interface CredentialDisplay {
+  name: string
+  schema: string
+  styles: CredentialDefinition["styles"]
+  display:
+    | {
+        properties: DisplayVal[]
+      }
+    | { [k: string]: DisplayVal }
```

Comment:
The `properties` field should always be defined. The rest of the fields i.e. `title`, `subtitle`, etc, can in addition to that.

Reply:
one way it works is like this. Otherwise there are issues with indexing
```
  display: {
    properties: DisplayVal[]
    title?: DisplayVal
    subtitle?: DisplayVal
    description?: DisplayVal
    [x: string]: any
  }

```

Reply:
what do you mean by "issues with indexing"?

Reply:
Ok updated!